src/encoded/static/components/search.js

```diff
@@ -974,39 +977,27 @@ TextFilter.propTypes = {
 // Displays the entire list of facets. It contains a number of <Facet> cmoponents.
 /* eslint-disable react/prefer-stateless-function */
 export class FacetList extends React.Component {
-    
     shouldComponentUpdate(nextProps) {
-        console.log("should the facet list update?");
-        console.log(!_.isEqual(this.props, nextProps));
         return !_.isEqual(this.props, nextProps);
     }
-    
+
     render() {
         const { context, facets, filters, mode, orientation, hideTextFilter, addClasses, modifyFacetsFlag } = this.props;
-        
-        let facetsModified = facets;
-        if (modifyFacetsFlag){
-            console.log("we are modifying the facets for regulome results");
-            // eliminate facets that we do not want to enable search for
-            facetsModified = facets.filter( facet => (facet.field !== 'files.file_type' && facet.field !=='status' && facet.field !== 'replicates.library.biosample.donor.organism.scientific_name' && facet.field !== 'assembly' && facet.field !== 'annotation_type'));
-        
-            // this is an extremely long-winded approach but my more abbreviated attempts did not work
-            facetsModified[0].terms = facets[0].terms;
-            facets[1].terms.forEach(function(f){
-                let flag = 0;
-                facets[0].terms.forEach(function(ff){
-                    if (f === ff){
-                        flag = 1;
-                    }
-                });
-                if (flag === 0){
-                    f.linkKey = 'annotation_type';
-                    facetsModified[0].terms.push(f);
-                }
-            });
-        
-            facetsModified[0].title = 'Method';
-            facetsModified[1].title = 'Biosample';
+
+        const facetsModified = facets;
+        if (modifyFacetsFlag) {
+            const methodIndex = facetsModified.map(facet => facet.title).indexOf('Assay');
```

Comment:
Could change this from .map().indexOf() to [.findIndex()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) so it can be done in one step, and also .findIndex() stops after it finds a match where .map() goes through the whole array no matter what.